resources/modes/print/print.js

```diff
@@ -73,6 +84,7 @@ $(function() {
 
       // With nothing in the queue, start autopaint!
       if (cncserver.state.buffer.length === 0) {
+        $("#processing-dialog").show()
```

Comment:
This is the right place for the logic to check to see if we should show the dialog!... Too bad it doesn't have any logic yet ;)

In pseudo JS code this would be something along the lines of: 
```
// Check every 3 seconds if there's anything in the buffer and we're still running.
var idleTimer = setInterval(checkIdle, 3000);
  
function checkIdle() {
  if (buffer.length == 0 && !paused && !completed)  {
    $("#processing-dialog").show();
  } else {
    $("#processing-dialog").hide();
  }
}
```
Then it'd probably be a good idea inside of the complete callback to put `clearInterval(idleTimer);`, which destroys the repeat timer for you. You might need to shuffle the scope of some vars here of course.